File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
class Solution {
    public int solution(int[] nums) {
        int n = nums.length;
        boolean[] visited = new boolean[200001];
      	int count = 0;
        
        for (int i=0; i<n; i++) {
            int cur = nums[i];
            if (!visited[cur]) {
                count++;
                visited[cur] = true;
            }
        }
        
        return Math.min(n / 2, count);
    }
    
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    const visited = new Set();
    let count = 0;
    
    for (const num of nums) {
        if (!visited.has(num)) {
            visited.add(num);
            count++;
        }
    }
    
    return Math.min(count, nums.length/2);
}